### PR TITLE
Update babel targets

### PR DESCRIPTION
**Pull request by turt2live**
**Review with https://github.com/matrix-org/matrix-react-sdk/pull/3957**

### CODE REVIEW

**Comment by turt2live** [comment]
I don't know why the riot tests blew up here

File: babel.config.js [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
+                    "last 2 Chrome versions", "last 2 Firefox versions", "last 2 Safari versions"
+                ]
+            },
+        }],
+        "@babel/preset-typescript",
+        "@babel/preset-flow",
+        "@babel/preset-react"
+    ],
+    "plugins": [
+        ["@babel/plugin-proposal-decorators", {"legacy": false, decoratorsBeforeExport: true}],
```
Comment (jryans) [inline babel.config.js]: Can we omit the `legacy: false` bit since that's the default?

File: babel.config.js [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-    "@babel/plugin-proposal-object-rest-spread",
-    "@babel/plugin-transform-flow-comments",
-    "@babel/plugin-syntax-dynamic-import",
-    "@babel/plugin-transform-runtime"
-  ]
+    "sourceMaps": true,
+    "presets": [
+        ["@babel/preset-env", {
+            "targets": {
+                "browsers": [
```
Comment (dbkr) [inline babel.config.js]: Fwiw they're removing this soon in favour of just setting 'targets' (https://babeljs.io/docs/en/babel-preset-env#targetsbrowsers) - we may as well do it now.

File: babel.config.js (hunk 1 above)
Comment (dbkr) [inline babel.config.js]: presumably making them not legacy fixes the crash? do we understand why?

also it looks like the proposal is now at stage 2 with export before decorators, so I think we're on the losing side at this point. We might want to give in & use the other syntax before we add them to every component.

File: babel.config.js (hunk 2 above)
Comment (turt2live) [reply babel.config.js]: so our `modules` is at the wrong level?

File: babel.config.js (hunk 1 above)
Comment (turt2live) [reply babel.config.js]: They're constructed completely differently in the transpiled code. Decorators are indeed stage 2, but also doing it not-legacy means we're closer to the TypeScript version which can work more easily.

The syntax is the same for usage, just different for definition. Though again, it's now closer to TypeScript's version.

**Comment by turt2live** [comment]
Tests failing due to https://github.com/vector-im/riot-web/issues/12109 as far as I can tell.

**Pull request merged**